impl/src/metadata.rs

```diff
@@ -19,16 +19,58 @@ use std::{
   string::String,
 };
 
-use anyhow::{anyhow, Result};
+use anyhow::{anyhow, Context, Result};
 
+use cargo_lock::Lockfile;
 use cargo_metadata::MetadataCommand;
-pub use cargo_metadata::{DependencyKind, Metadata, Node, Package, PackageId};
 
+use glob::glob;
+use pathdiff::diff_paths;
+use regex::Regex;
+use rustc_serialize::hex::ToHex;
 use tempfile::TempDir;
+use url::Url;
 
-use rustc_serialize::hex::ToHex;
+pub use cargo_metadata::{DependencyKind, Metadata, Node, Package, PackageId};
 
 const SYSTEM_CARGO_BIN_PATH: &str = "cargo";
+pub(crate) const DEFAULT_CRATE_REGISTRY_URL: &str = "https://crates.io";
+pub(crate) const DEFAULT_CRATE_INDEX_URL: &str = "https://github.com/rust-lang/crates.io-index";
+
+#[derive(Debug, Clone)]
+pub struct RazeMetadata {
+  pub metadata: Metadata,
+  pub checksums: HashMap<String, String>,
+  pub workspace_root: PathBuf,
+  pub lockfile: Option<Lockfile>,
+}
+
+impl RazeMetadata {
+  pub fn new(
```

Comment:
Ignored this comment?

Reply:
I don't feel strongly about this, but I'm really surprised you're not seeing these comments. I just click the `Files Changed` tab. Perhaps you're not expanding the unexpanded files?

Reply:
`checksums` was not pub but I can make it so. I thought keeping it private would drive users to use `get_checksum` instead of directly accessing the field but I suppose it's not that valuable.